ocs_ci/ocs/ui/base_ui.py

```diff
@@ -0,0 +1,178 @@
+        element = wait.until(ec.element_to_be_clickable((type, by_locator)))
+        return True if element.get_attribute("aria-expanded") == "true" else False
+
+    def choose_expanded_mode(self, mode, by_locator, type):
+        """
+        Select the element mode (expanded or collapsed)
+
+        by_locator (str): GUI element needs to operate on.
+        type (By): Set of supported locator strategies
+        mode (bool): True if element expended, False otherwise
+
+        """
+        current_mode = self.is_expanded(by_locator=by_locator, type=type)
+        if mode != current_mode:
+            self.do_click(by_locator=by_locator, type=type)
+
+    def take_screenshot(self):
+        """
+        Take screenshot using python code
+
+        """
+        time.sleep(2)
+        self.cnt_screenshot += 1
+        filename = os.path.join(
+            self.screenshots_folder, str(self.cnt_screenshot).zfill(3)
+        )
+        self.driver.save_screenshot(filename)
+
+
+def login_ui():
+    """
+    Login to OpenShift Console
+
+    return:
+        driver (Selenium WebDriver)
+
+    """
+    logger.info("Get URL of OCP console")
+    console_url = run_cmd(
+        "oc get consoles.config.openshift.io cluster -o"
+        "jsonpath='{.status.consoleURL}'"
+    )
+    logger.info("Get password of OCP console")
+    password = get_kubeadmin_password()
+    password = password.rstrip()
+    browser = ocsci_config.UI_SELENIUM.get("browser_type")
+    if browser == "chrome":
+        logger.info("chrome browser")
+        chrome_options = Options()
+
+        ignore_ssl = ocsci_config.UI_SELENIUM.get("ignore_ssl")
+        if ignore_ssl:
+            chrome_options.add_argument("--ignore-ssl-errors=yes")
+            chrome_options.add_argument("--ignore-certificate-errors")
+
+        # headless browsers are web browsers without a GUI
+        headless = ocsci_config.UI_SELENIUM.get("headless")
+        if headless:
+            chrome_options.add_argument("--headless")
+
+        chrome_browser_type = ocsci_config.UI_SELENIUM.get("chrome_type")
+        if chrome_browser_type == "chromium":
+            chrome_browser = ChromeType.CHROMIUM
+
+        driver = webdriver.Chrome(
+            ChromeDriverManager(chrome_type=chrome_browser).install(),
+            chrome_options=chrome_options,
+        )
+    if browser == "firefox":
+        logger.info("firefox browser")
+        driver = webdriver.Firefox()
+
+    wait = WebDriverWait(driver, 30)
+    driver.maximize_window()
+    driver.get(console_url)
+    element = wait.until(ec.element_to_be_clickable((By.ID, "inputUsername")))
+    element.send_keys("kubeadmin")
+    element = wait.until(ec.element_to_be_clickable((By.ID, "inputPassword")))
+    element.send_keys(password)
+    element = wait.until(
+        ec.element_to_be_clickable(
+            (By.XPATH, "/html/body/div/div/main/div/form/div[4]/button")
+        )
+    )
+    element.click()
```

Comment:
Why you are not using the previously prepared methods `do_send_keys` and `do_click` here?

Reply:
I use `close_browser` and `login_ui` functions as fixture (on conftest file)